pkg/configuration/configuration.go

```diff
@@ -298,6 +326,9 @@ func (ev *extendedViper) Set(key string, value interface{}) {
 	localStorage := ev.storage
 	isPersisted := ev.persistedKeys[key]
 	ev.viper.Set(key, value)
+	if ev.defaultCache != nil {
+		ev.defaultCache.Delete(key)
```

Comment:
I'm wondering if this is correct. Maybe it would be better to update the cache with the new value?